### PR TITLE
bsp: Update new API for buttons and display

### PR DESCRIPTION
**Pull request by espzav**
# ESP-BSP Pull Request checklist

- [x] Version of modified component bumped
- [x] CI passing

# Change description
Updated ESP-WROVER-KIT with new BSP API
- Added some missing APIs (e.g. SPIFFS)
- Updated some old APIs (e.g. buttons)
- Updated LCD API

### CODE REVIEW

File: esp_wrover_kit/priv_include/bsp_err_check.h [hunk 1]
```diff
@@ -46,12 +46,6 @@ extern "C" {
         }                                 \
     } while(0)
 
-#define BSP_NULL_CHECK(x, ret) do { \
-        if ((x) == NULL) {          \
-            return ret;             \
-        }                           \
-    } while(0)
-
```
Comment (tore-espressif) [inline esp_wrover_kit/priv_include/bsp_err_check.h:54]: Can we leave it here? just for consistency with other BSPs. Or does it trigger warnings?

File: esp_wrover_kit/idf_component.yml [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
+version: "1.5.0"
 description: Board Support Package for ESP-WROVER-KIT
 url: https://github.com/espressif/esp-bsp/tree/master/esp_wrover_kit
 
 targets:
     - esp32
 
 dependencies:
-  idf: ">=4.4" # esp_lcd component was added in v4.4
+  idf: ">=4.4.5"
```
Comment (tore-espressif) [inline esp_wrover_kit/idf_component.yml:9]: I'm sorry, but can you please remind me why do we need 4.4.5?

**Review by tore-espressif — APPROVED** [review]
Overall LGTM!
I didn't check on HW, though

File: esp_wrover_kit/idf_component.yml (hunk 2 above)
Comment (espzav) [reply esp_wrover_kit/idf_component.yml:9]: We need changes in `esp_lcd` (registering flush callback) for `esp_lvgl_port`. This feature was backported and released in version 4.4.5

File: esp_wrover_kit/priv_include/bsp_err_check.h (hunk 1 above)
Comment (espzav) [reply esp_wrover_kit/priv_include/bsp_err_check.h:54]: There was same macro twice. In all BSPs, one of them was removed.

**Pull request merged**